javascript/ql/src/semmle/javascript/security/dataflow/HardcodedCredentials.qll

```diff
@@ -20,5 +20,18 @@ module HardcodedCredentials {
     override predicate isSource(DataFlow::Node source) { source instanceof Source }
 
     override predicate isSink(DataFlow::Node sink) { sink instanceof Sink }
+
+    override predicate isAdditionalFlowStep(DataFlow::Node src, DataFlow::Node trg) {
+      exists(Base64::Encode encode | src = encode.getInput() and trg = encode.getOutput())
```

Comment:
Is Base64 the only encoding we care about in practice? Hex seems like an equally likely and unreasonable way to secure sensitive information.

(MD5 "encoding" and friends are fine here BTW, we have `js/weak-cryptographic-algorithm` for that)

Reply:
Base64 is more likely as it is mandated by HTTP Basic authentication:
> In basic HTTP authentication, a request contains a header field in the form of Authorization: Basic <credentials>, where credentials is the Base64 encoding of ID and password joined by a single colon :.

Reply:
I see. I need to read that RFC again :laughing: